pkgs/applications/misc/1password-gui/default.nix

```diff
@@ -87,7 +97,13 @@ stdenv.mkDerivation rec {
         --replace 'Exec=/opt/1Password/${pname}' 'Exec=${pname}'
 
       # Polkit file
-      install -Dm 0644 -t $out/share/polkit-1/actions com.1password.1Password.policy
+      if [ ! -z "$POLICY_OWNERS" ]
+      then
+        mkdir -p $out/share/polkit-1/actions
+        eval "cat <<EOF
```

Comment:
I think rather than this `eval`-`cat`-`cat` dance (which I assume is so that this `POLICY_OWNERS` variable gets interpolated into the template), I would prefer that it is `substituteInPlace`'d. Essentially, I'd apply this diff:

<details><summary><code>git</code> diff</summary>

```diff
diff --git a/pkgs/applications/misc/1password-gui/default.nix b/pkgs/applications/misc/1password-gui/default.nix
index 3a41a3630e2..bfed4f9427a 100644
--- a/pkgs/applications/misc/1password-gui/default.nix
+++ b/pkgs/applications/misc/1password-gui/default.nix
@@ -34,8 +34,12 @@
   # The 1Password polkit file requires a list of users for whom polkit
   # integrations should be enabled. This should be a list of strings that
   # correspond to usernames.
-, polkit_users ? []
+, polkitPolicyOwners ? [ ]
 }:
+let
+  # Convert the polkit_users variable to a polkit-compatible string for the polkit file.
+  policyOwners = lib.concatStringsSep " " (map (user: "unix-user:${user}") polkitPolicyOwners);
+in
 stdenv.mkDerivation rec {
   pname = "1password";
   version = "8.6.0";
@@ -51,11 +55,6 @@ stdenv.mkDerivation rec {
   dontBuild = true;
   dontPatchELF = true;
 
-  # Convert the polkit_users variable to a polkit-compatible string for the polkit file.
-  POLICY_OWNERS = builtins.foldl' (a: b: if (a == "") then b else (a + " " + b)) "" (
-    builtins.map (val: "unix-user:" + val) polkit_users
-  );
-
   installPhase =
     let rpath = lib.makeLibraryPath [
       alsa-lib
@@ -95,16 +94,13 @@ stdenv.mkDerivation rec {
       install -Dt $out/share/applications resources/${pname}.desktop
       substituteInPlace $out/share/applications/${pname}.desktop \
         --replace 'Exec=/opt/1Password/${pname}' 'Exec=${pname}'
-
-      # Polkit file
-      if [ ! -z "$POLICY_OWNERS" ]
-      then
+    '' + lib.optionalString (polkitPolicyOwners != [ ])
+      ''
+        # Polkit file
         mkdir -p $out/share/polkit-1/actions
-        eval "cat <<EOF
-$(cat ./com.1password.1Password.policy.tpl)
-EOF" > $out/share/polkit-1/actions/com.1password.1Password.policy
-      fi
-
+        substitute com.1password.1Password.policy.tpl $out/share/polkit-1/actions/com.1password.1Password.policy \
+          --replace "\''${POLICY_OWNERS}" "${policyOwners}"
+      '' + ''
       # Icons
       cp -a resources/icons $out/share
 

```

</details>

It's much easier to see what's going on, IMHO.

I also:

* changed the `polkit_users` variable name so that it's more representative of its use (`polkitPolicyOwners`)
* simplified the `policyOwners` list using `lib` functions
* moved the `policyOwners` out of the environment
* used `lib.optionalString` to conditionally create the policy file instead of bash

---

Feel free to accept as many (or as little) of these changes as you like but I feel very strongly about avoiding the `eval`-`cat`-`cat` dance.

Reply:
These look great! I'm going to apply all of them.

Reply:
Done!